packages/fast-figma-plugin-msft/src/figma/node.ts

```diff
@@ -0,0 +1,225 @@
+            return children;
+        } else {
+            return [];
+        }
+    }
+
+    public supports(): Array<RecipeTypes | "designSystem"> {
+        return Object.keys(RecipeTypes)
+            .concat("designSystem")
+            .filter(key => {
+                switch (key) {
+                    case RecipeTypes.backgroundFills:
+                    case RecipeTypes.strokeFills:
+                    case "designSystem":
+                        return [
+                            isFrameNode,
+                            isRectangleNode,
+                            isPolygonNode,
+                            isStarNode,
+                            isComponentNode,
+                            isInstanceNode,
+                        ].some((test: (node: BaseNode) => boolean) => test(this.node));
+                    case RecipeTypes.foregroundFills:
+                        return isTextNode(this.node);
+                    default:
+                        return false;
+                }
+            }) as Array<RecipeTypes | "designSystem">;
+    }
+
+    public paint(data: RecipeData): void {
+        switch (data.type) {
+            case RecipeTypes.strokeFills:
+            case RecipeTypes.backgroundFills:
+            case RecipeTypes.foregroundFills:
+                this.paintColor(data);
+                break;
+            default:
+                throw new Error(`Recipe could not be painted ${JSON.stringify(data)}`);
+        }
+    }
+
+    public parent(): FigmaPluginNode | null {
+        const parent = this.node.parent;
+
+        if (parent === null) {
+            return null;
+        }
+
+        return new FigmaPluginNode(parent.id);
+    }
+
+    public getEffectiveBackgroundColor(): ColorRGBA64 {
+        let node: BaseNode | null = this.node;
+
+        while (node !== null) {
+            if ((node as GeometryMixin).fills) {
+                const fills = (node as GeometryMixin).fills;
+
+                if (Array.isArray(fills)) {
+                    const paints: SolidPaint[] = fills.filter(
+                        (fill: Paint) => fill.type === "SOLID"
+                    );
+
+                    /**
+                     * TODO: how do we process multiple paints?
+                     */
+                    if (paints.length === 1) {
+                        const parsed = ColorRGBA64.fromObject(paints[0].color);
+                        if (parsed instanceof ColorRGBA64) {
+                            return parsed;
+                        }
+                    }
+                }
+            }
+
+            node = node.parent;
+        }
+
+        return new ColorRGBA64(1, 1, 1);
+    }
+
+    protected getPluginData<K extends keyof PluginNodeData>(key: K): PluginNodeData[K] {
+        try {
+            return JSON.parse(this.node.getPluginData(key as string));
+        } catch (e) {
+            return key === "designSystem" ? ({} as any) : []; // Why does keyof not work here?!
```

Comment:
did you mean to leave this comment in? lol

Reply:
+1 and "I don't know!" :)

Reply:
I mean... kinda. I'd like this to work but it doesn't

Reply:
Looks good then 👍